### ISSUE TITLE
[httpd-remoteip.conf] Possible error in private IP space

### ISSUE THREAD

**Issue by bonfilsd** [cloudfoundry/php-buildpack#205]
Hello,

With our team, we have noticed a possible error in httpd-remoteip.conf file 
* since v4.2.0 : https://github.com/cloudfoundry/php-buildpack/blob/v4.2.0/defaults/config/httpd/extra/httpd-remoteip.conf#L5
* until current / master branch v4.3.30 : https://github.com/cloudfoundry/php-buildpack/blob/v4.3.30/defaults/config/httpd/extra/httpd-remoteip.conf#L5

So, in according to https://tools.ietf.org/html/rfc1918, in '3. Private Address Space' section, private IP addresses are : 

```pre
10.0.0.0        -   10.255.255.255  (10/8 prefix)
172.16.0.0      -   172.31.255.255  (172.16/12 prefix)
192.168.0.0     -   192.168.255.255 (192.168/16 prefix)
```
So, we have 172.16.0.0/12 in place of 176.16.0.0/12

In addition, we have add localhost IP address (127.0.0.1).

So, we have overload the file like this : 

```apache
#
# Adjust IP Address based on header set by proxy
#
<IfModule remoteip_module>
    RemoteIpHeader x-forwarded-for
    RemoteIpInternalProxy 127.0.0.0/8 10.0.0.0/8 172.16.0.0/12 192.168.0.0/16
</IfModule>

#
# Set HTTPS environment variable if we came in over secure
#  channel.
<IfModule setenvif_module>
    SetEnvIf x-forwarded-proto https HTTPS=on
</IfModule>
```

Thanks in advance for your answer.

Enjoy ;)

**Comment by sclevine** [comment]
Definitely looks like a bug to me. Prioritized a story to fix it.

**Comment by dmikusa-pivotal** [comment]
@bonfilsd - Why did you add 127.0.0.0/8?  Those aren't "remote" IPs.  Is there a situation where your app deployed to CF is receiving a x-forwarded-for header with a 127.0.0.0/8 IP in it?  

I can't think of a way that would legitimately happen.  On CF the x-forwarded-for header should really only have the gorouter IP and any of your external load balancer IPs. None of those should put a 127.0.0.0/8 IP in that header.

**Comment by bonfilsd** [comment]
@dmikusa-pivotal - you're right. After some tests it's appear that localhost IP address is useless.

Finally, we have a file like this :

```apache
#
# Adjust IP Address based on header set by proxy
#
<IfModule remoteip_module>
    RemoteIpHeader x-forwarded-for
    RemoteIpInternalProxy 10.0.0.0/8 172.16.0.0/12 192.168.0.0/16
</IfModule>

#
# Set HTTPS environment variable if we came in over secure
#  channel.
<IfModule setenvif_module>
    SetEnvIf x-forwarded-proto https HTTPS=on
</IfModule>
```

Thks ;)

**Comment by bonfilsd** [comment]
Hi,

why you don't have added tags <IfModule> ... </IfModule> in the fix ?
https://github.com/cloudfoundry/php-buildpack/commit/97bda3a80b1810c4076d276a7bb134227b3a96a8

++

**Comment by sclevine** [comment]
I'm guessing it's because those modules are always present.
@dmikusa-pivotal thoughts?

**Comment by dmikusa-pivotal** [comment]
+1 - The build pack will always install those modules.

**Comment by bonfilsd** [comment]
Ok but it's simply for prevention and best practices
However as you want / decide.

**Comment by dmikusa-pivotal** [comment]
Thinking about this a little more, I think it's good we don't have the IfModule tags for the build pack.  This configuration is required for apps running on CF, because you're always going to be behind at least one layer of proxies.  

If the module doesn't load for some reason and we had IfModule tags, HTTPD would silently skip over this and just not configure it.  That could in turn cause unexpected problems with the app, for example breaking enforcement of using HTTPS.  I think it's better if we don't have the IfModule tags, so HTTPD will blow up and indicate there's a problem.  This shouldn't ever happen, but if for some strange reason there was a bug we'd find it more quickly and could fix it.

**Comment by bonfilsd** [comment]
Ok I understand needs for "visibility / debug" purpose.
Thanks for your explanations / precisions.

Have a good day.

**Comment by sclevine** [comment]
Fixed on develop: 97bda3a80b1810c4076d276a7bb134227b3a96a8
